src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -118,6 +124,24 @@ public function __construct(ClassMetadataFactoryInterface $classMetadataFactory
         }
         $this->classDiscriminatorResolver = $classDiscriminatorResolver;
         $this->objectClassResolver = $objectClassResolver;
+
+        if (null === $instantiator) {
+            $instantiator = new Instantiator($classMetadataFactory, $this->classDiscriminatorResolver, $propertyTypeExtractor, null, $nameConverter);
+
+            if ($this->denormalizer instanceof DenormalizerInterface) {
+                $instantiator->setDenormalizer($this->denormalizer);
+            }
+        }
+        $this->instantiator = $instantiator;
+    }
+
+    public function setDenormalizer(DenormalizerInterface $denormalizer)
```

Comment:
Can't we just reuse the existing `SerializerAwareInterface` infrastructure and check that the injected serializer is an instance of `DenormalizerInterface`? I feel like it's introducing a lot of complexity that may not worth it.

Reply:
I actually don't see how I could do this since `Instantiator` is not part of normalizers/encoders ?